### PR TITLE
test: clean up detox build/run scripts

### PR DESCRIPTION
**Pull request by cortisiko**
## **Description**

The purpose of this PR is to simplify the build and run commands for running tests locally. 

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -528,7 +528,7 @@ workflows:
                 node -v
                 export METAMASK_ENVIRONMENT='local'
                 export METAMASK_BUILD_TYPE='main'
-                IGNORE_BOXLOGS_DEVELOPMENT="true" FORCE_BUNDLING=true yarn test:e2e:android:bitrise:build
```
Comment (cortisiko) [inline bitrise.yml:531]: no need to have a specific build step for bitrise. The specific build step should be either for debug or release builds

File: bitrise.yml [hunk 2]
```diff
@@ -601,7 +601,7 @@ workflows:
                 fi
                 export METAMASK_ENVIRONMENT='local'
                 export METAMASK_BUILD_TYPE='main'
-                IGNORE_BOXLOGS_DEVELOPMENT="true" FORCE_BUNDLING=true yarn test:e2e:android:bitrise:run "$TEST_SUITE_FOLDER" --testNamePattern="$TEST_SUITE"
```
Comment (cortisiko) [inline bitrise.yml:604]: no need to have a specific build step for bitrise. The specific build step should be either for debug or release builds

File: docs/readme/testing.md [hunk 3]
```diff
@@ -41,18 +41,31 @@ yarn watch
 
 You can trigger the tests against a `release` or `debug` build. It recommended that you trigger the tests against a debug build.
 
-To trigger the tests on a debug build run this command:
+To build the app for testing on an iOS debug build run this command:
```
Comment (cortisiko) [inline docs/readme/testing.md:44]: Updated readme to reflect changes

File: docs/readme/testing.md [hunk 4]
```diff
@@ -74,23 +87,23 @@ If you have already built the application for Detox and want to run a specific t
 For iOS
 
 ```bash
-yarn test:e2e:ios:debug:single e2e/specs/TEST_NAME.spec.js
+yarn test:e2e:ios:debug:run e2e/specs/TEST_NAME.spec.js
```
Comment (cortisiko) [inline docs/readme/testing.md:90]: Updated readme to reflect changes

File: package.json [hunk 5]
```diff
@@ -57,14 +57,10 @@
     "test:unit:update": "time jest -u ./app/",
     "test:e2e": "yarn test:e2e:ios && yarn test:e2e:android",
     "test:e2e:ios": "detox build -c ios.sim.release && detox test -c ios.sim.release",
-    "test:e2e:ios:debug": "IS_TEST='true' detox build -c ios.sim.debug && detox test -c ios.sim.debug",
```
Comment (cortisiko) [inline package.json:60]: We had a separate build script for building the debug version of the app for testing. I am cleaning up the scripts

**Review by Cal-L — APPROVED** [review]
LGTM

**Comment by leotm** [comment]
just realised we should update [notion](https://www.notion.so/metamask-consensys/Mobile-E2E-Testing-Detox-Documentation-5cf672bf346f4b9bb4c92188ba2f6c14) too, or could be better instead to refer to [testing.md](https://github.com/MetaMask/metamask-mobile/blob/main/docs/readme/testing.md) in _Running tests > Locally_ section for less maintainance

**Pull request merged**